lib/active_record/events/method_factory.rb

```diff
@@ -48,7 +48,11 @@ def define_inverse_predicate_method(module_, naming)
 
       def define_action_method(module_, naming)
         module_.send(:define_method, naming.action) do
-          touch(naming.field)
+          if persisted?
+            touch(naming.field)
+          else
+            send("#{naming.field}=", Time.zone.now)
```

Comment:
Also, let's use `Time.current` instead of `Time.zone.now` :)

Reply:
I updated this to be:

```ruby
self[naming.field] = current_time_from_proper_timezone
```

The reason I used `Time.zone.now` is because [Rubocop prefers it](https://www.rubydoc.info/gems/rubocop/0.57.2/RuboCop/Cop/Rails/TimeZone), but I switched to `current_time_from_proper_timezone` as that's how rails does it:

https://github.com/rails/rails/blob/f855139f3d2bb9b032613279d0adfbd6a77a2d07/activerecord/lib/active_record/persistence.rb#L912